### PR TITLE
Fix List Associations integ tests

### PR DESCRIPTION
**Pull request by stisac**
*Issue #, if available:* https://sim.amazon.com/issues/AML-106487

*Description of changes:* Fix the list association tests after removing the skips

*Testing done:*

tox -e py39 -- tests/unit/sagemaker/lineage

collected 45 items

tests/unit/sagemaker/lineage/test_action.py ........... [ 24%]
tests/unit/sagemaker/lineage/test_artifact.py ............ [ 51%]
tests/unit/sagemaker/lineage/test_association.py ....... [ 66%]
tests/unit/sagemaker/lineage/test_context.py ........... [ 91%]
tests/unit/sagemaker/lineage/test_dataset_artifact.py . [ 93%]
tests/unit/sagemaker/lineage/test_endpoint_context.py . [ 95%]
tests/unit/sagemaker/lineage/test_model_artifact.py . [ 97%]
tests/unit/sagemaker/lineage/test_visualizer.py . [100%]

-> tox -e py39 -- tests/integ/sagemaker/lineage


collected 29 items                                                                                                                                                                                                                        

tests/integ/sagemaker/lineage/test_action.py .......                                                                                                                                                                                [ 24%]
tests/integ/sagemaker/lineage/test_artifact.py ........                                                                                                                                                                             [ 51%]
tests/integ/sagemaker/lineage/test_association.py ....                                                                                                                                                                              [ 65%]
tests/integ/sagemaker/lineage/test_context.py .......                                                                                                                                                                               [ 89%]
tests/integ/sagemaker/lineage/test_dataset_artifact.py .                                                                                                                                                                            [ 93%]
tests/integ/sagemaker/lineage/test_endpoint_context.py .                                                                                                                                                                            [ 96%]
tests/integ/sagemaker/lineage/test_model_artifact.py .                                         [100%]

### CODE REVIEW

**Comment by stisac** [comment]
The failure of the sagemaker-python-sdk-pr test is unrelated to the changes made in this revision

File: tests/integ/sagemaker/lineage/test_association.py [hunk 1]
```diff
@@ -35,27 +32,51 @@ def test_list(association_objs, sagemaker_session):
...
     tag = {"Key": "foo", "Value": "bar"}
     association_obj.set_tag(tag)
-    assert association_obj.get_tag() == tag
+
+    while True:
+        actual_tags = sagemaker_session.sagemaker_client.list_tags(
+            ResourceArn=association_obj.source_arn
+        )["Tags"]
+        if actual_tags:
+            break
```
Comment (danabens) [inline tests/integ/sagemaker/lineage/test_association.py:66]: lets put a sleep in here - don't want to spam list_tags from a test

File: tests/integ/sagemaker/lineage/test_association.py [hunk 2]
```diff
@@ -35,27 +32,51 @@ def test_list(association_objs, sagemaker_session):
...
+def test_tags(association_obj, sagemaker_session):
+    tags = [{"Key": "foo1", "Value": "bar1"}]
+    association_obj.set_tags(tags)
+
+    while True:
+        actual_tags = sagemaker_session.sagemaker_client.list_tags(
+            ResourceArn=association_obj.source_arn
+        )["Tags"]
+        if actual_tags:
+            break
```
Comment (danabens) [inline tests/integ/sagemaker/lineage/test_association.py:80]: add sleep

**Comment by danabens** [comment]
```
...
.tox/py38/lib/python3.8/site-packages/docker/models/images.py:287: BuildError
```

looks good, just add wait

File: tests/integ/sagemaker/lineage/test_association.py (hunk 1 above)
Comment (stisac) [reply tests/integ/sagemaker/lineage/test_association.py:66]: Added, Thank you

File: tests/integ/sagemaker/lineage/test_association.py (hunk 2 above)
Comment (stisac) [reply tests/integ/sagemaker/lineage/test_association.py:80]: Added Thank you

**Comment by stisac** [comment]
> ```
> ...
> .tox/py38/lib/python3.8/site-packages/docker/models/images.py:287: BuildError
> ```
> 
> looks good, just add wait/sleep

Sure, Thank you

**Review by danabens — APPROVED** [review]

**Review by ahsan-z-khan — APPROVED** [review]

**Pull request merged**